From: Chris Wright <chrisw@sous-sol.org>
Date: Mon, 8 Feb 2010 20:28:02 -0500
Subject: [vhost] vhost-net: defer f->private_data until setup succeeds
Message-id: <20100208202802.GH17153@redhat.com>
Patchwork-id: 23203
O-Subject: [RHEL6 PATCH 7/8] vhost-net: defer f->private_data until setup
	succeeds
Bugzilla: 562837
RH-Acked-by: Herbert Xu <herbert.xu@redhat.com>
RH-Acked-by: Juan Quintela <quintela@redhat.com>

BZ#562837
https://bugzilla.redhat.com/show_bug.cgi?id=562837

Description
Trivial change, just for readability.  The filp is not installed on
failure, so the current code is not incorrect (also vhost_dev_init
currently has no failure case).  This just treats setting f->private_data
as something with global scope (sure, true only after fd_install).

Signed-off-by: Chris Wright <chrisw@redhat.com>

Upstream Status of the patches
This has been merged upstream in a single commit
3a4d5c94e959359ece6d6b55045c3f046677f55c

Brew build#: 2247495

Test Status:
it has been tested by customers (IBM), and in house on x86 and x86-64
platforms.

Signed-off-by: Aristeu Rozanski <arozansk@redhat.com>

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index 51025a1..4c89283 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -326,7 +326,6 @@ static int vhost_net_open(struct inode *inode, struct file *f)
 	int r;
 	if (!n)
 		return -ENOMEM;
-	f->private_data = n;
 	n->vqs[VHOST_NET_VQ_TX].handle_kick = handle_tx_kick;
 	n->vqs[VHOST_NET_VQ_RX].handle_kick = handle_rx_kick;
 	r = vhost_dev_init(&n->dev, n->vqs, VHOST_NET_VQ_MAX);
@@ -338,6 +337,9 @@ static int vhost_net_open(struct inode *inode, struct file *f)
 	vhost_poll_init(n->poll + VHOST_NET_VQ_TX, handle_tx_net, POLLOUT);
 	vhost_poll_init(n->poll + VHOST_NET_VQ_RX, handle_rx_net, POLLIN);
 	n->tx_poll_state = VHOST_NET_POLL_DISABLED;
+
+	f->private_data = n;
+
 	return 0;
 }
 
